### PR TITLE
feat: improve registration patterns docs

### LINKED ISSUE #5669: [Docs] Improve registration patterns docs
### Description

Questions related to this topic lead to think that expanding and updating the [Registering workflows](https://docs.flyte.org/en/latest/flyte_fundamentals/registering_workflows.html) section would be useful to make sure it's easier to understand.

[Question](https://discuss.flyte.org/t/22925077/hi-again-wave-we-ve-been-going-around-in-circles-with-this-o)
[Question](https://flyte-org.slack.com/archives/CREL4QVAQ/p1724062662953219)

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

### PR DESCRIPTION
**Pull request by siiddhantt**
## Closes #5669 
- Improved `registering_workflows` docs with updated and new note for issues and workaround

### Screenshots
Screenshot 2024-10-04 at 7 16 08 PM

Screenshot 2024-10-04 at 7 16 22 PM

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

File: docs/flyte_fundamentals/registering_workflows.md [hunk 1]
```diff
@@ -345,6 +352,15 @@ a {py:class}`~flytekit.remote.remote.FlyteRemote` object. You can learn more
 about how to do this {ref}`here <flytekit:design-control-plane>`.
 ```
 
+```{note}
+When using `pyflyte` to run jobs, the `k8sPod.dataConfig` section is automatically added to the task configuration. This section is crucial to ensure the task has access to the necessary input files.
```
Comment (davidmirror-ops) [inline docs/flyte_fundamentals/registering_workflows.md:356]: Thanks! I know this was on the issue description but given that this is a solved issue I think we can skip it altogether. It's useful but doesn't age well.

File: docs/flyte_fundamentals/registering_workflows.md [hunk 2]
```diff
@@ -297,6 +296,14 @@ pyflyte --pkgs <dir1> --pkgs <dir2> package ...
 This is useful in cases where you want to register two different Flyte projects
 that you maintain in a single place.
 
+
+If you encounter a ``ModuleNotFoundError`` when packaging, ensure to include the correct source paths by using the `--source` option. For instance:
+
+```{prompt} bash $
+pyflyte --pkgs <dir1> package --source ./src -f
```
Comment (davidmirror-ops) [inline docs/flyte_fundamentals/registering_workflows.md:303]: @wild-endeavor 
I know the "old" flags still work but wanted to double-check that this command would still be valid even if we deprecate the old flags? (re https://github.com/flyteorg/flytekit/pull/2690)

File: docs/flyte_fundamentals/registering_workflows.md (hunk 1 above)
Comment (siiddhantt) [reply docs/flyte_fundamentals/registering_workflows.md]: Got it! Will remove it altogether.

**Comment by neverett** [comment]
Hi @siiddhantt the file you're updating has been moved to [/docs/user_guide/flyte_fundamentals/registering_workflows.md](https://github.com/flyteorg/flyte/blob/master/docs/user_guide/flyte_fundamentals/registering_workflows.md) -- could you merge master into your branch and apply your changes to the doc in the new location? Thanks!

**Comment by siiddhantt** [comment]
hi @neverett , merged master into this branch, hope it's good now, thanks!

File: docs/user_guide/flyte_fundamentals/registering_workflows.md [hunk 3]
```diff
@@ -297,6 +297,14 @@ pyflyte --pkgs <dir1> --pkgs <dir2> package ...
 This is useful in cases where you want to register two different Flyte projects
 that you maintain in a single place.
 
+
+If you encounter a ``ModuleNotFoundError`` when packaging, ensure to include the correct source paths by using the `--source` option. For instance:
```
Comment (neverett) [inline docs/user_guide/flyte_fundamentals/registering_workflows.md:301]:
```suggestion
If you encounter a ``ModuleNotFoundError`` when packaging, use the `--source` option to include the correct source paths. For instance:
```

File: docs/user_guide/flyte_fundamentals/registering_workflows.md [hunk 4]
```diff
@@ -297,6 +297,14 @@ pyflyte --pkgs <dir1> --pkgs <dir2> package ...
...
+
+```{prompt} bash $
+pyflyte --pkgs <dir1> package --source ./src -f
+```
+
+This addition helps in avoiding the module detection issue and ensures smooth registration and execution of tasks.
```
Comment (neverett) [inline docs/user_guide/flyte_fundamentals/registering_workflows.md:307]: I think this line can be removed, since it doesn't add any new information.

**Review by neverett — DISMISSED** [review]
I left a few suggestions, but otherwise this looks good to me. Thanks!

**Review by neverett — APPROVED** [review]

**Pull request merged**